tests/rados/test_bilog_trim.py

```diff
@@ -0,0 +1,143 @@
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    #CEPH-83575437
+    #BZ-2056818
+    This test is to verify no issues are observed during deep-scrub when rgw bilog
+    trimming occurs with OSD down
+    Steps:
+    Ref:
+    - https://bugzilla.redhat.com/show_bug.cgi?id=2056818#c42
+    - https://bugzilla.redhat.com/show_bug.cgi?id=2056818#c50
+    1. Deploy minimal RGW multisite setup
+    2. Add data to both the sites in S3 buckets and let the minimal client IO running
+    3. Keep running the deep-scrub for index pool PGs
+    4. Set noout flag
+    5. stop the secondary OSD on the primary site(site1)
+    6. run the following command
+       radosgw-admin bilog autotrim
+    7. Once the above command finishes, start the primary OSD that was stopped in step 4
+    8. Retrigger the deep-scrub for eight PGs in index pool
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+
+    try:
+        # set noout flag
+        if not rados_utils.configure_osd_flag(ceph_cluster, action="set", flag="noout"):
+            log.error("Could not set noout flag on the cluster")
+            raise Exception("Could not set noout flag on the cluster")
+
+        # get acting set for the rgw index pool
+        _pool_name = "primary.rgw.buckets.index"
+        acting_set = rados_obj.get_pg_acting_set(pool_name=_pool_name)
+        log.info(f"Acting set for pool {_pool_name}: {acting_set}")
+
+        # secondary osd of 1st PG from rgw index pool
+        second_osd = acting_set[1]
+
+        # get list on active OSDs
+        osd_list = rados_obj.get_osd_list(status="up")
+
+        # bring down 30% of total UP OSDs
+        osd_down = random.choices(osd_list, k=floor(0.3 * len(osd_list)))
+        osd_down.append(second_osd)
+
+        log.info(f"List of OSD to be stopped: {osd_down}")
+
+        # trigger deep-scrub on the rgw index pool
+        rados_obj.run_deep_scrub(pool=_pool_name)
+
+        for osd_id in osd_down:
+            if not rados_obj.change_osd_state(action="stop", target=int(osd_id)):
+                log.error(f"Could not stop OSD.{osd_id}")
+                raise Exception(f"Could not stop OSD.{osd_id}")
+
+        # for a duration of 7 mins, trigger bi-log trimming along with deep-scrub
+        timeout_time = datetime.datetime.now() + datetime.timedelta(seconds=420)
+        while datetime.datetime.now() < timeout_time:
+            cephadm.shell(args=["radosgw-admin bilog autotrim"])
+            rados_obj.run_deep_scrub(pool=_pool_name)
+            time.sleep(30)
+
+        # start the stopped OSDs
+        for osd_id in osd_down:
+            if not rados_obj.change_osd_state(action="start", target=int(osd_id)):
+                log.error(f"Could not start OSD.{osd_id}")
+                raise Exception(f"Could not start OSD.{osd_id}")
+
```

Comment:
Should we wait for cluster to get health_ok before we move to next steps? like wait for clean PGs?

Reply:
Added code to wait for clean PGs but not failing the test if the cluster fails to reach active+clean within 600 secs